R/find_anaemia.R

```diff
@@ -0,0 +1,316 @@
+                            (df$age > 15 & df$sex == 1) &
+                            (df$hb >= 80  & df$hb < 110 ), 2, anaemia_all)
+  }
+  ##
+  if(!is.null(age) & !is.na(hb) & !is.na(sex)) {
+    anaemia_all <- ifelse(!is.na(df$age) & !is.na(df$hb) & !is.na(df$sex) &
+                            (df$age > 15 & df$sex == 1) &
+                            (df$hb < 80 ), 3, anaemia_all)
+  }
+  ##
+  # child U5 category
+  if(!is.null(age) & !is.na(hb)) {
+    anaemia_all <- ifelse(!is.na(df$age) & !is.na(df$hb) &
+                            (df$age < 5) &
+                            (df$hb >= 100  & df$hb < 110 ), 1, anaemia_all)
+  }
+  ##
+  if(!is.null(age) & !is.na(hb)) {
+    anaemia_all <- ifelse(!is.na(df$age) & !is.na(df$hb) &
+                            (df$age < 5) &
+                            (df$hb >= 70  & df$hb < 100 ), 2, anaemia_all)
+  }
+  ##
+  if(!is.null(age) & !is.na(hb)) {
+    anaemia_all <- ifelse(!is.na(df$age) & !is.na(df$hb) &
+                            (df$age < 5) &
+                            (df$hb < 70 ), 3, anaemia_all)
+  }
+  ##
+  # child 5 - 11 category
+  if(!is.null(age) & !is.na(hb)) {
+    anaemia_all <- ifelse(!is.na(df$age) & !is.na(df$hb) &
+                            (df$age >= 5 & df$age < 12) &
+                            (df$hb >= 110  & df$hb < 115 ), 1, anaemia_all)
+  }
+  ##
+  if(!is.null(age) & !is.na(hb)) {
+    anaemia_all <- ifelse(!is.na(df$age) & !is.na(df$hb) &
+                            (df$age >= 5 & df$age < 12) &
+                            (df$hb >= 80  & df$hb < 110 ), 2, anaemia_all)
+  }
+  ##
+  if(!is.null(age) & !is.na(hb)) {
+    anaemia_all <- ifelse(!is.na(df$age) & !is.na(df$hb) &
+                            (df$age >= 5 & df$age < 12) &
+                            (df$hb < 80 ), 3, anaemia_all)
+  }
+  ##
+  # child 12 - 14 category
+  if(!is.null(age) & !is.na(hb)) {
+    anaemia_all <- ifelse(!is.na(df$age) & !is.na(df$hb) &
+                            (df$age >= 12 & df$age < 15) &
+                            (df$hb >= 110  & df$hb < 120 ), 1, anaemia_all)
+  }
+  ##
+  if(!is.null(age) & !is.na(hb)) {
+    anaemia_all <- ifelse(!is.na(df$age) & !is.na(df$hb) &
+                            (df$age >= 12 & df$age < 15) &
+                            (df$hb >= 80  & df$hb < 110 ), 2, anaemia_all)
+  }
+  ##
+  if(!is.null(age) & !is.na(hb)) {
+    anaemia_all <- ifelse(!is.na(df$age) & !is.na(df$hb) &
+                            (df$age >= 12 & df$age < 15) &
+                            (df$hb < 80 ), 3, anaemia_all)
+  }
+  ##
+  if(add) {
+    df$anaemia_all <- anaemia_all
+    anaemia        <- df
+  }
+
+  return(anaemia)
+}
+
+
+
+################################################################################
+# alternated function using cut
+################################################################################
+
+################################################################################
+#
```

Comment:
Description of the function should indicate that supplied values for Hb are already cleaned.